client/web/compose/src/components/PageBlocks/RecordConfigurator.vue

```diff
@@ -168,9 +202,35 @@ export default {
     addRuleDisabled () {
       return this.block.options.fields.filter(f => !f.isRequired).length === this.block.options.fieldConditions.length
     },
+
+    recordSelectorFields () {
+      return this.module.fields.filter(f => f.kind === 'Record')
+    },
+
+    fieldModule () {
+      return (this.options.referenceField && this.referenceModule) ? this.referenceModule : this.module
+    },
+  },
+
+  watch: {
+    'options.referenceModuleID': {
+      immediate: true,
+      handler (id) {
+        if (id) {
+          this.findModuleByID({ namespace: this.namespace.namespaceID, moduleID: this.options.referenceModuleID })
```

Comment:
This should be done on created, since otherwise you're doing it twice (in updateReferenceModule)